src/auth/session.service.ts

```diff
@@ -57,14 +59,18 @@ export class SessionService {
     // The refresh token is valid for refreshTokenValidSeconds seconds.
     // By default, it is valid for defaultRefreshTokenValidSeconds seconds.
     refreshTokenValidSeconds?: number,
+    sessionValidSeconds?: number,
   ): Promise<string> {
     const session = new Session();
     session.userId = userId;
     session.authorization = JSON.stringify(authorization);
     if (refreshTokenValidSeconds == null) {
       refreshTokenValidSeconds = this.defaultRefreshTokenValidSeconds;
```

Comment:
Rejected. This is not supported by TS grammar.

You can run the following code online at https://www.jyshare.com/compile/5577/:
```typescript
function and(a: string, b?: string = 'b'): string {
	return a + b;
}
console.log(and('1', '2'))
```

It will tell you:
script.ts(1,25): error TS1015: Parameter cannot have question mark and initializer.

Reply:
That's why I say `refreshTokenValidSeconds = this.defaultRefreshTokenValidSeconds`
In your case, `b: string = 'b'` is accepted, which still means that the `b` is optional and given default value 'b' if not provided.

Reply:
fixed.
https://github.com/SageSeekerSociety/cheese-backend/commit/2699f4c03774cc0165f1607900913c783dfd9569

Reply:
The grammar you mentioned works.
```typescript
function and(a: string, b: string = 'b'): string {
	return a + b;
}
console.log(and('1', '2'))
console.log(and('1'))
console.log(and('1', undefined))
```
gives output
```
12
1b
1b
```

I thought the default value gammar is something static before. It's not. Thanks you